tacs/problems/static.py

```diff
@@ -754,86 +896,217 @@ def solve(self, Fext=None):
+
+            # Set initnorm as the norm of rhs
+            self.initNorm = np.real(self.rhs.norm())
 
-        # Get rhs vector
-        self.K.mult(self.u, self.rhs)
-        self.rhs.axpy(-1.0, self.res)
+            # Starting Norm for this computation
+            self.startNorm = np.real(self.res.norm())
 
-        # Set initnorm as the norm of rhs
-        self.initNorm = np.real(self.rhs.norm())
+            initNormTime = time.time()
 
-        # Starting Norm for this computation
-        self.startNorm = np.real(self.res.norm())
+            # Solve Linear System for the update
+            hasConverged = self._solveLinear(self.res, self.update)
 
-        initNormTime = time.time()
+            self.update.scale(-1.0)
 
-        # Solve Linear System for the update
-        success = self.KSM.solve(self.res, self.update)
+            solveTime = time.time()
+
+            # Update State Variables
+            self.assembler.getVariables(self.u)
+            self.u.axpy(1.0, self.update)
+            self.assembler.setVariables(self.u)
+
+            stateUpdateTime = time.time()
+
+            # Get updated residual
+            self.getResidual(self.res, Fext)
+            self.finalNorm = np.real(self.res.norm())
+
+            finalNormTime = time.time()
+
+            # If timing was was requested print it, if the solution is nonlinear
+            # print this information automatically if prinititerations was requested.
+            if self.getOption("printTiming"):
+                self._pp("+--------------------------------------------------+")
+                self._pp("|")
+                self._pp("| TACS Solve Times:")
+                self._pp("|")
+                self._pp(
+                    "| %-30s: %10.3f sec"
+                    % ("TACS Setup Time", setupProblemTime - startTime)
+                )
+                self._pp(
+                    "| %-30s: %10.3f sec"
+                    % ("TACS Solve Init Time", initSolveTime - setupProblemTime)
+                )
+                self._pp(
+                    "| %-30s: %10.3f sec"
+                    % ("TACS Init Norm Time", initNormTime - initSolveTime)
+                )
+                self._pp(
+                    "| %-30s: %10.3f sec"
+                    % ("TACS Solve Time", solveTime - initNormTime)
+                )
+                self._pp(
+                    "| %-30s: %10.3f sec"
+                    % ("TACS State Update Time", stateUpdateTime - solveTime)
+                )
+                self._pp(
+                    "| %-30s: %10.3f sec"
+                    % ("TACS Final Norm Time", finalNormTime - stateUpdateTime)
+                )
+                self._pp("|")
+                self._pp(
+                    "| %-30s: %10.3f sec"
+                    % ("TACS Total Solution Time", finalNormTime - startTime)
+                )
+                self._pp("+--------------------------------------------------+")
+
```

Comment:
@timryanb do you mind if I remove some of the timing categories here so that we can print the same output for linear and nonlinear solves?

I'd suggest we just keep the solve init time and the solve time as everything else seems very minor.

Reply:
That's fine by me